core/src/main/java/com/exadel/etoolbox/linkinspector/core/services/resolvers/InternalLinkResolverImpl.java

```diff
@@ -107,7 +107,10 @@ public void validate(Result result, ResourceResolver resourceResolver) {
         if (status.getCode() == HttpStatus.SC_NOT_FOUND && StringUtils.isNotBlank(internalLinksHost)) {
             String prefix = StringUtils.startsWithAny(internalLinksHost, HTTP_SCHEMA, HTTPS_SCHEMA) ? EMPTY : HTTPS_SCHEMA;
             String origin = StringUtils.stripEnd(internalLinksHost, "/");
-            externalLinkResolver.validate(new LinkResult(result.getType(), prefix + origin + result.getValue()), resourceResolver);
+            String extension = StringUtils.isAlpha(StringUtils.substringAfterLast(result.getValue(), ".")) ? StringUtils.EMPTY : ".html";
```

Comment:
Sooo... no `.mp4` videos, `.woff2` fonts, `.7z`archives, etc. -? ))
I would recommend using one of the OOTB url parsers. Probably `org.apache.sling.api.uri.SlingUriBuilder.parse(result.getValue(), resourceResolver).getExtension()`.

If not possible, `RegExp("\\.\\w+$").matcher(result.getValue()).find()`

Reply:
thank you, done